source/common/modules/persistent-data-container/index.ts

```diff
@@ -139,6 +139,7 @@ export default class PersistentDataContainer {
     try {
       const content = await fs.readFile(this._filePath, { encoding: 'utf-8' })
 
+      if (content.trim().length === 0) throw new Error("Container at " + this._filePath + " was empty (content is null).");
```

Comment:
This could be a good addition to catch instances where the initializer found a perfectly normal file, and something caused the file to be emptied out in the meantime. Just two suggestions:

1. I tried to stop using shorthand `if`-statements across the entire codebase, and surround everything with `{` and `}` now
2. Instead of using string concatenation, please use template strings (with backticks and `${variable}`)